Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM python:3.12.3-alpine as base
+    RUN pip install poetry
+    WORKDIR /app
+    ENTRYPOINT ["poetry", "run", "flask", "run", "--host=0.0.0.0"]
+    COPY poetry.lock ./
+    COPY pyproject.toml ./
+    RUN poetry install
+
+FROM base as development
+    ENV FLASK_DEBUG=true
+    EXPOSE 5000
+
+FROM base as production
+    ENV FLASK_DEBUG=false
+    COPY todo_app ./todo_app
```

Comment:
Since you only COPY this folder it's not a massive issue that there's  no `.dockerignore`, but I'd still recommend one to help you future-proof. At the very least, including `.env` to cover your largest security risk (building and sharing an image with secrets baked in) and `**/__pycache` to cover folders under `todo_app` feels like a reasonable minimal version